compiler/src/dotty/tools/dotc/parsing/JavaParsers.scala

```diff
@@ -907,4 +907,28 @@ object JavaParsers {
       unit
     }
   }
+
+
+  /**
+   * OutlineJavaParser parses top-level declarations in `source` to find declared classes, ignoring their bodies (which
+   * must only have balanced braces). This is used to map class names to defining sources.
+   * This is necessary even for Java, because the filename defining a non-public classes cannot be determined from the
+   * classname alone.
+   */
+  class OutlineJavaParser(source: SourceFile)(implicit ctx: Context) extends JavaParser(source) {
+
+    def skipBraces[T](body: T): T = {
```

Comment:
This is identical to the Scala version except for the lack of XMLSTART handling, can't we just use the Scala version for Java too ?